### PR TITLE
Porting UxPerf Workloads

### PR DESCRIPTION
**Pull request by marcbonnici**

### CODE REVIEW

File: wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java [hunk 1]
```diff
@@ -173,6 +215,108 @@ public int getDisplayCentreHeight() {
...
+                view.swipeRight(steps);
+                break;
+            case NULL:
+                throw new Exception("No direction specified");
+            default:
+                break;
+        }
+    }
+
+        public void uiObjectVertPinchIn(UiObject view, int steps, int percent) throws Exception {
```
Comment (setrofim) [inline wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java]: Indentation level seems to change from this line onwards.

File: wa/framework/workload.py [hunk 2]
```diff
@@ -218,7 +229,25 @@ def teardown(self, context):
...
-        pass
+        self.remove_assets(context)
+
+    def deploy_assets(self, context, target_directory=None):
+        if not self.deployable_assets:
+            return
+
+        if not target_directory:
+            target_directory = self.target.working_directory
+        self.target.execute('mkdir -p {}'.format(target_directory))
```
Comment (setrofim) [inline wa/framework/workload.py]: Why is this necessary? Shouldn't the working_directory already exist by the time we get to initializing workloads?

File: wa/framework/workload.py [hunk 3]
```diff
@@ -218,7 +229,25 @@ def teardown(self, context):
...
+
+        for i, asset in enumerate(self.deployable_assets):
+            self.target.push(self.asset_files[i], target_directory)
+            self.deployed_assets.append(self.target.path.join(target_directory,
```
Comment (setrofim) [inline wa/framework/workload.py]: The join could be replaced with `self.target.get_workpath(asset)`

File: wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java (hunk 1 above)
Comment (marcbonnici) [reply wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java]: I can't seem to see the change in indention. Has this been fixed?

File: wa/framework/workload.py (hunk 2 above)
Comment (marcbonnici) [reply wa/framework/workload.py]: The idea here was that you could set a custom directory for the assets to be deployed to instead of the working directory so the custom folder may not have existed. However I've re-done this bit to be part of the base Workload Class and to use an attribute to store the directory path instead.

File: wa/framework/workload.py (hunk 3 above)
Comment (marcbonnici) [reply wa/framework/workload.py]: This was left as join to allow for custom directories.

File: wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java (hunk 1 above)
Comment (setrofim) [reply wa/framework/uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java]: yup, seems to be fixed now. :+1:
